src/devices/machine/mc68328.h

```diff
@@ -920,10 +809,437 @@ class mc68328_device : public m68000_device
 	devcb_write8     m_out_ld_cb;
 	lcd_info_changed_delegate m_lcd_info_changed_cb;
 
-	static const uint32_t VCO_DIVISORS[8];
+	static const u32 VCO_DIVISORS[8];
+};
+
+class mc68328_device : public mc68328_base_device
+{
+public:
+	mc68328_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	template <int Line> auto out_port_j() { return m_out_port_j_cb[Line].bind(); }
+	template <int Line> auto out_port_k() { return m_out_port_k_cb[Line].bind(); }
+	template <int Line> auto out_port_m() { return m_out_port_m_cb[Line].bind(); }
+	template <int Line> auto in_port_j() { return m_in_port_j_cb[Line].bind(); }
+	template <int Line> auto in_port_k() { return m_in_port_k_cb[Line].bind(); }
+	template <int Line> auto in_port_m() { return m_in_port_m_cb[Line].bind(); }
+
+private:
+	// device-level overrides
+	virtual void device_resolve_objects() override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
```

Comment:
Not overly important, but these overridden protected members should be protected.